proto/gnmi_ext/gnmi_ext.proto

```diff
@@ -159,3 +160,30 @@ message Depth {
   // returned.
   uint32 level = 1;
 }
+
+// ConfigSubscription extension allows clients to subscribe to configuration
+// schema nodes only.
+message ConfigSubscription {
+  oneof action {
+    // ConfigSubscriptionStart is sent by the server in the SubscribeRequest
```

Comment:
The documentation in the use cases doesn't seem to correspond to this, and it seems like there's overloading of these fields.

a) the client sends something from the target with `ConfigSubscriptionStart` set. I suggest this is either moved to the `SubscriptionList` or we specifically call this something that is clearer that we are filtering the contents to config only nodes. 

b) I'd suggest making this a separate extension -- or making this `oneof` separate client->target and target->client fields.

Reply:
This looks like a typo in the comment, it should be `ConfigSubscriptionStart is sent by the client in the SubscribeRequest`

It's true that receiving `ConfigSubscriptionStart` really means two things for the target:
1) Send the content of config only nodes.
2) Send a ConfigSubscriptionSyncDone (for on-change subscriptions) after the notifications of each commit are streamed.

If we add a field (called `type` like `GetRequest` ?) under SubscriptionList that can be set to: `BOTH`, `CONFIG` or `STATE`, should the presence of this extension imply that `type` is set to `CONFIG` ?

Reply:
IMHO, if we add a field to the `SubscriptionList` then we should consider receiving this extension without the type being set to `CONFIG` is an error. Having APIs where something optional (an extension) implies the value of something required is error prone and creates additional testing overhead [0] -- whereas if we're explicit, it's clear what is meant.


[0]: for example, if this extension implies `CONFIG` then you will end up with the case whereby if I sent something with `STATE` to a target that doesn't support the extension (and therefore ignores it, validly) it will return a different set of paths to a target that does understand the extension and has implicitly meant that I assumed `CONFIG`.

Reply:
THis looks addressed to me.  @robshakir ?

Reply:
@karimra do you want to add a extension field to the `SubscriptionList` indicating config vs. state?   Seems this will give the client an explicit indication that the target is or is not honoring the `ConfigSubscriptionStart`.

Another way to handle this might be to implement a way to get the supported extensions?  Then a client can programmatically know if `ConfigSubscription` is supported.

Reply:
Yes, I was planning to do that in a separate PR if that's ok. 
My understanding is that it won't be an extension but an additional field called `type` with 3 possible values `BOTH`, `CONFIG` and `STATE`.

Reply:
That's my proposal, yes. 

I think (reading this proto and the spec) that there are two distinct things that are being covered in this extension:

1. the ability to subscribe to configuration only values, because the consumer is not interested in state data.
2. being able to have a configuration subscription that has some form of barrier semantics that allows a client to start a subscription and understand where in the stream a particular commit completes.

For 1., my view is that this sounds generally applicable -- a `CONFIG` subscription would be defined as any writeable value, a `STATE` one would be any read-only value, and `BOTH`/`ALL` will be all leaves (and the default for backwards compatibility).

For 2., I think this is where an extension is a good way to go.

If we agree, we can close this comment.

Reply:
We agree, I will open a PR for 1.

Reply:
thank you!